### PR TITLE
change ?= to an ifeq in the Makefiles for issue #284

### PR DESCRIPTION
**Pull request by arjenve**
Convert all ?= lines in the Makefiles of the codes (which will not work if the variable is set but empty) into 3 lines with ifeq.

### CODE REVIEW

**Comment by ipelupessy** [comment]
I think this is overkill (makes a lot of Makefiles ugly to read)..better to fix few cases where it matters (if it ever does, simplex case I don't think qualifies because the root cause was the config failing (which could be set by hand))

in any case: there are errors in the conversion...

File: src/amuse/community/phiGRAPE/Makefile [hunk 1]
```diff
@@ -24,35 +36,58 @@ CODELIB=$(BUILDDIR)/libphigrape.a
...
 
-SAPPORO_LIBDIRS ?= -L$(AMUSE_DIR)/lib/sapporo_light
-OPENMP_CFLAGS ?= 
-SAPPORO_LIBS ?= $(SAPPORO_LIBDIRS) -lsapporo $(OPENMP_CFLAGS)
-GRAPE6_LIBDIRS ?= -L$(GRAPE6)/lib
+ifeq ($(strip $(SAPPORO_LIBDIRS)),)
+    SAPPORO_LIBDIRS = -L$(AMUSE_DIR)/lib/sapporo_light
+endif
+ifeq ($(strip $(OPENMP_CFLAGS)),)
+    OPENMP_CFLAGS = SAPPORO_LIBS ?= $(SAPPORO_LIBDIRS) -lsapporo $(OPENMP_CFLAGS)
```
Comment (ipelupessy) [inline src/amuse/community/phiGRAPE/Makefile]: this doesn't look right...

**Comment by ipelupessy** [comment]
I think the better alternative is to not export unset variables in config.mk, or to check whether variables that should be set, are....

(really, the resulting makefiles are clearly not written by/for humans..)

**Comment by arjenve** [comment]
Well yes, the configure could not find the libs. And if you then change the Makefile in testing you get very hard to understand and debug behavior. (I.E. your change will not be used for an unknown reason)

File: src/amuse/community/phiGRAPE/Makefile (hunk 1 above)
Comment (arjenve) [reply src/amuse/community/phiGRAPE/Makefile]: Yes I see, clearly not right! (also original had
 OPENMP_CFLAGS ?=
So optional not setting, which is very strange)

I'll update this one

**Comment by ipelupessy** [comment]
also I am not sure if the original behaviour is in many cases not the exact desired behaviour, e.g. some flag being empty can be the desired contents of the flag...

**Comment by arjenve** [comment]
Well yes, that's also true. Lets bin this pull request.

**Comment by ipelupessy** [comment]
ok, lets do that;-)

**Pull request closed without merging**